autonomy/chain/utils.py

```diff
@@ -0,0 +1,98 @@
+# -*- coding: utf-8 -*-
+# ------------------------------------------------------------------------------
+#
+#   Copyright 2023 Valory AG
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+
+"""Utility functions."""
+from json import JSONDecodeError
+from typing import Dict, List
+
+from aea.configurations.base import PackageConfiguration
+from aea.configurations.data_types import PublicId
+from aea.crypto.base import LedgerApi
+from requests import get as r_get
+from requests.exceptions import ConnectionError as RequestConnectionError
+
+from autonomy.chain.base import RegistryContracts
+from autonomy.chain.exceptions import DependencyError, FailedToRetrieveComponentMetadata
+from autonomy.chain.metadata import IPFS_URI_PREFIX
+
+
+def get_ipfs_hash_from_uri(uri: str) -> str:
+    """Split IPFS hash from the ipfs uri"""
+
+    return uri.replace(IPFS_URI_PREFIX, "")
+
+
+def resolve_component_id(
+    ledger_api: LedgerApi,
+    contract_address: str,
+    token_id: int,
+) -> Dict:
+    """Resolve component ID"""
+
+    metadata_uri = RegistryContracts.component_registry().get_token_uri(
+        ledger_api=ledger_api,
+        contract_address=contract_address,
+        token_id=token_id,
+    )
+
+    try:
+        return r_get(url=metadata_uri).json()
+    except (RequestConnectionError, JSONDecodeError) as e:
+        raise FailedToRetrieveComponentMetadata() from e
+
+
+def verify_component_dependencies(
+    ledger_api: LedgerApi,
+    contract_address: str,
+    dependencies: List[int],
```

Comment:
So we need to manually provide this list? We need to think how we can improve this experience